app/src/main/cpp/skyline/common.h

```diff
@@ -338,6 +339,35 @@ namespace skyline {
         constexpr std::size_t Hash(std::string_view view) {
             return frz::elsa<frz::string>{}(frz::string(view.data(), view.size()), 0);
         }
+
+        /**
+         * @brief Returns the most appropriate integral type and count to represent the given T object
```

Comment:
I'd actually recommend changing this a bit and naming it `PointerFor` (also change `type` to a pointer type) and explicitly stating that it selects the largest possible pointer type for representing an object alongside providing the size of the object in terms of the underlying type.